### PR TITLE
CP-3323 BACKPATCH Disallow retry of canceled requests

### PR DESCRIPTION
**Pull request by evanweible-wf**
Same as #243 but backpatched to the 2.x line.

@Workiva/web-platform-pp

### CODE REVIEW

**Comment by aviary2-wf** [comment]
## Raven
Number of Findings: 0

**Comment by evanweible-wf** [comment]
Ignore the smithy failures here - smithy wasn't set up for this repo until 3.0.0.

File: .travis.yml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 language: dart
 dart:
-  - stable
+  - 1.19.1
```
Comment (sebastianmalysa-wf) [inline .travis.yml]: Just curious, why use this version of dart?

Comment (evanweible-wf) [inline .travis.yml]: There's an issue with coverage on more recent versions that I'll need to get fixed in dart_dev.

**Comment by maxwellpeterson-wf** [comment]
+1

**Comment by sebastianmalysa-wf** [comment]
+1

**Comment by evanweible-wf** [comment]
@jayudey-wf ready for you

**Comment by jayudey-wf** [comment]
QA Resource Approval: +10

- [ ] Testing instruction
- [ ] Dev +1's
- [ ] Dev/QA +10 with detail of what was tested
  - passing CI
- [ ] Unit test created/updated
- [ ] All unit tests pass 

Merging into master.

**Comment by teresarevious-wf** [comment]
RM +1

 verified dependencies

**Pull request merged**